Geometry_Engine/Compute/SortAlongCurve.cs

```diff
@@ -209,6 +209,9 @@ public static List<Point> SortAlongCurve(this List<Point> points, Polyline curve
             else if (tmpResult.Count == 1)
                 return tmpResult;
 
+            // For a close polyline there will be a duplicate point which needs to be removed
+            if (tmpResult.Count > points.Count - 1)
+                tmpResult = tmpResult.Distinct().ToList();
```

Comment:
Is it safer to check only if the last point is the same as the first point for BHoM closed polylines? I wouldn't expect there to be more points along the curve than input but maybe there is a case for that?

Reply:
Have updated to check `Curve.IsClosed()` and tested for open and closed `Polyline`. 

The `points` are not always the same as the `ControlPoints` from the `Polyline`.